src/query/query_eval.rs

```diff
@@ -69,88 +71,70 @@ fn eval_brackets<'a>(
 fn eval_test<'a>(
     index: &'a Index,
     test: Test,
-    value: Literal,
-    lookup_field: Lookup,
+    lhs: TestValue,
+    rhs: TestValue,
 ) -> Result<Vec<&'a Packet>, QueryError> {
     Ok(index
         .packets
         .iter()
-        .filter(|packet| lookup_filter(packet, &test, &value, &lookup_field))
+        .filter(|packet| lookup_filter(packet, &test, &lhs, &rhs))
         .collect())
 }
 
-fn lookup_filter(packet: &Packet, test: &Test, value: &Literal, lookup: &Lookup) -> bool {
-    match (test, lookup) {
-        (Test::Equal, Lookup::Id) => match value {
-            Literal::String(str) => packet.id == *str,
-            _ => false
-        },
-        (Test::NotEqual, Lookup::Id) => match value {
-            Literal::String(str) => packet.id != *str,
-            _ => false
-        },
-        (_, Lookup::Id) => false,
-        (Test::Equal, Lookup::Name) => match value {
-            Literal::String(str) => packet.name == *str,
-            _ => false
-        },
-        (Test::NotEqual, Lookup::Name) => match value {
-            Literal::String(str) => packet.name != *str,
-            _ => false
-        },
-        (_, Lookup::Name) => false,
-        (test, Lookup::Parameter(param_name)) => packet.test_parameter(param_name, test, value),
+fn lookup_filter(packet: &Packet, test: &Test, lhs: &TestValue, rhs: &TestValue) -> bool {
+    let lhs_literal = match lhs {
+        TestValue::Literal(value) => Some(value.clone()),
+        TestValue::Lookup(lookup) => packet.lookup_value(lookup)
+    };
+    let rhs_literal = match rhs {
+        TestValue::Literal(value) => Some(value.clone()),
+        TestValue::Lookup(lookup) => packet.lookup_value(lookup)
+    };
```

Comment:
Tiny bit of repeated logic, is it worth pulling out into its own fn?

Reply:
Thanks, I've tidied that up